src/org/rascalmpl/uri/URIUtil.java

```diff
@@ -443,6 +433,32 @@ public static ISourceLocation removeOffset(ISourceLocation prev) {
     public static ISourceLocation createFromURI(String value) throws URISyntaxException {
         return vf.sourceLocation(createFromEncoded(value));
     }
+
+	/**
+	 * Extracts the extension (the characters after the last . in the file name),
+	 * if any, and otherwise returns the empty string.
+	 * @param loc
+	 * @return
+	 */
+	public static String getExtension(ISourceLocation loc) {
+		String path = loc.getPath();
+		boolean endsWithSlash = path.endsWith(URIUtil.URI_PATH_SEPARATOR);
+		if (endsWithSlash) {
+			path = path.substring(0, path.length() - 1);
+		}
+
+		if (path.length() > 1) {
+			int slashIndex = path.lastIndexOf(URIUtil.URI_PATH_SEPARATOR);
+			int index = path.substring(slashIndex).lastIndexOf('.');
```

Comment:
nit: we can also use the `fromIndex` parameter of the `lastIndex` function, to avoid a substring, (and avoid the math on line 455)

Reply:
Ok